### PR TITLE
Updated to go1.23

### PR DESCRIPTION
**Pull request by lmnzx**
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

**Review by abhishek9686 — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       - name: Setup go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.19
+          go-version-file: go.mod
```
Comment (abhishek9686) [inline .github/workflows/test.yml:32]: Are quotes not required around go.mod?

Comment (lmnzx) [reply .github/workflows/test.yml:32]: they are optional, i will still add them as to maintain the code style

**Review by abhishek9686 — APPROVED** [review]

File: docker/Dockerfile-go-builder
Comment (abhishek9686) [inline docker/Dockerfile-go-builder:1]: we should follow versioned tag here https://github.com/gravitl/netmaker/blob/76027a6115103ea38258632aa26f72fe5ba1d09b/Dockerfile#L2

we can push go builder image with tag 1.23.0

Comment (lmnzx) [reply docker/Dockerfile-go-builder:1]: sorry, i over looked it, fixing it

**Review by abhishek9686 — APPROVED** [review]

File: docker/Dockerfile-go-builder
Comment (abhishek9686) [reply docker/Dockerfile-go-builder:1]: https://github.com/gravitl/netmaker/blob/76027a6115103ea38258632aa26f72fe5ba1d09b/.github/workflows/docker-builder.yml#L29 tag in this workflow is pointing to latest, also seems like this is running in push to develop always which is unnecessary

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**